app/services/invoices/create_service.rb

```diff
@@ -178,13 +177,19 @@ def create_credit_note_credit(invoice)
     end
 
     def create_coupon_credit(invoice)
-      credit_result = Credits::AppliedCouponService.new(
-        invoice: invoice,
-        applied_coupon: applied_coupon,
-      ).create
-      credit_result.throw_error unless credit_result.success?
+      applied_coupons.each do |applied_coupon|
+        return unless invoice.amount_cents&.positive?
```

Comment:
Lint/NonLocalExitFromIterator: Non-local exit from iterator, without return value. `next`, `break`, `Array#find`, `Array#any?`, etc. is preferred.